lib/faraday/adapter/net_http.rb

```diff
@@ -42,8 +42,9 @@ def initialize(app = nil, opts = {}, &block)
 
       def build_connection(env)
         net_http_connection(env).tap do |http|
-          http.use_ssl = env[:url].scheme == 'https' if http.respond_to?(:use_ssl=)
-          configure_ssl(http, env[:ssl])
+          if env[:url].scheme == 'https' && env[:ssl]
```

Comment:
Minor: What if this is even smaller? Making the smallest check first.

```suggestion
          if env[:ssl] && env[:url].scheme == 'https'
```

Reply:
But, it's not important!

Reply:
@olleolleolle Thank you so much!